cetloxblock/static/js/services.js

```diff
@@ -0,0 +1,202 @@
+      }
+      return "prod";
+    }
+    return domainsMap[getEnv()];
+  }
+  var CetDomain = this.CetDomain = this.getCetDomain();
+
+  var PRODUCT_PLAYER = "//productplayer" + CetDomain + "/";
+  var PRODUCT_PLAYER_API = "//productplayer" + CetDomain + "/api/ApiProxy/";
+  var PRODUCT_PLAYER_SERVICES = "//productplayerservices" + CetDomain + "/";
+  var TREE_SERVICE = "//treeservice" + CetDomain + "/api/Trees/";
+  var TREE_LMS_SERVICE = "//treeservice" + CetDomain + "/api/TreeLMS/";
+  var DEFAULT_CET_COURSE = "Edx-adaptive-project";
+
+  this.getCourseEdxName = function () {
+    var match = /-v\d+:([^\/\+]+\+[^\/\+]+\+[^\/\+]+)[\/\+]/gi.exec(location.pathname);
+    if (match.length) {
+      return match[1];
+    }
+  }
+
+  function getCourseCetName(edxName) {
+    var url = PRODUCT_PLAYER + "edxCourses.json";
+    return $.Deferred(function (dfd) {
+      $.get(url).done(function (response) {
+        var courseObj = response[edxName];
+        if (courseObj) {
+          dfd.resolve(courseObj.path);
+        }
+        else {
+          //dfd.reject("Cet course is not defined");
+          dfd.resolve(DEFAULT_CET_COURSE);
+        }
+      }).fail(function (response) {
+        //dfd.reject("Failed to retrieve Cet course name. Error: " + response);
+        dfd.resolve(DEFAULT_CET_COURSE);
+      });
+    });
+  }
+
+  this.getCourseSettings = function (edxName) {
+    edxName = edxName || this.getCourseEdxName();
+    return $.Deferred(function (dfd) {
+      getCourseCetName(edxName)
+        .done(function (cetName) {
+          var url = PRODUCT_PLAYER_SERVICES + "GetProductConfigurationByName/" + cetName;
+
+          $.get(url)
+            .done(function (response) {
+              var settingsObj = {
+                edxName: edxName,
+                cetName: response.productName,
+                productId: response.productId,
+                folderId: response.rootDocumentId,
+                language: response.loLanguage,
+                cetDomain: CetDomain,
+              };
+              dfd.resolve(settingsObj);
+            })
+            .fail(function (response) {
+              dfd.reject("Failed to retrieve Cet course properties. Error: " + response);
+            });
+        })
+        .fail(function (error) {
+          dfd.reject(error);
+        });
+    });
+  }
+
+  // #region STUDIO SERVICES
+  this.getCourseItemList = function (courseSettings) {
+    var _this = this;
+    if (!courseSettings) {
+      return $.Deferred(function (dfd) {
+        this.getCourseSettings()
+        .done(function (settings) {
+          _this.getCourseItemList(settings).done(function (itemList) {
+            dfd.resolve(itemList);
+          }).fail(function (response) {
+            dfd.reject(response);
+          });
+        }).fail(function (response) {
+          dfd.reject(response);
+        });
+      });
+    }
+
+
+    // http://treeservice.cet.ac.il/api/Trees/a2326f9f-c094-4458-95ea-e832d7273e9c/he

```

Comment:
These example URLs are really useful for me to understand the data this code is processing!  Thanks for including them.